app/Generators/SuperHero.php

```diff
@@ -24,19 +20,41 @@ public function get( $post ) {
 
         $category = $post->terms( 'category' );
 
-        return Timber::compile(
+        return \Timber::compile(
             __DIR__ . '/views/super-hero.twig',
             [
-                'site' => new TimberSite(),
+                'site' => new \TimberSite(),
                 'post' => $post,
                 'post_category' => $post->post_category,
                 'post_category_slug' => $category[0]->slug,
                 'post_date' => gmdate('d M Y', strtotime($post->post_date)),
                 'SEGMENT_WRITE_KEY' => getenv( 'SEGMENT_WRITE_KEY' ),
-                'canonical_url' => $url
+                'canonical_url' => $url,
+                'adverts' => $this->get_adverts( $post )
             ],
             false,
-            TimberLoader::CACHE_NONE
+            \TimberLoader::CACHE_NONE
         );
     }
+
+    private function get_adverts( $post ) {
+        $domain = getenv('WEB_BASE_URL');
+
+        $category = get_the_category($post->ID);
+        $zone = '';
+
+        if ( $category[0] instanceof \WP_Term ) {
```

Comment:
What happens if the Post does not have a category? Would it trigger error `Undefined offset: 0`? Or does WP automatically add the category `Uncategorized`?